src/JitInterface/src/CorInfoImpl.cs

```diff
@@ -698,7 +698,12 @@ private uint getMethodAttribsInternal(MethodDesc method)
             if (method.IsSynchronized)
                 result |= CorInfoFlag.CORINFO_FLG_SYNCH;
             if (method.IsIntrinsic)
-                result |= CorInfoFlag.CORINFO_FLG_INTRINSIC;
+            {
+                if (getIntrinsicID(method, null) != CorInfoIntrinsics.CORINFO_INTRINSIC_Illegal)
```

Comment:
I don't particularly like that we have to do this, but RyuJIT asserts and dies if we report `CORINFO_FLG_JIT_INTRINSIC` on something that has an intrinsic ID.

@AndyAyersMS do we really need two flags here? Would a single flag, followed by a call made by RyuJIT to `getIntrinsicID` be enough to disambiguate?

Reply:
We could probably make do with just one flag, though I think the current model has some merit.

We're anticipating the arrival of hundreds of the new style intrinsics as the HW intrinsic work ramps up, so I'd prefer not to force the jit to call back through the interface each time to figure out which kind of intrinsic is in play. 

In CoreCLR these two kinds of intrinsics are distinct in the runtime, and the jit treats them differently, so it is fairly natural to have two flags, and the jit avoids a bit of extra work as the runtime already knows what kind of intrinsic this is.

In CoreRT there is evidently just one notion of an intrinsic in the runtime, and so one side or the other has to do this extra check (the runtime right now, the jit if we went down to just one flag), and it's probably slightly more efficient to do it in the runtime.

Reply:
Can the JIT be fixed to deal gracefully with both flags set, and not assert if the JIT/EE interface reports `CORINFO_FLG_JIT_INTRINSIC` on something that has an intrinsic ID?

Reply:
https://github.com/dotnet/coreclr/pull/14566

Reply:
It looks like the jit currently calls `getIntrinsicID` for both kinds of intrinsic. Not sure why I left it that way. So fixing the jit to sort things out when both flags are set seems ok.

Reply:
I have appended a commit to call getIntrinsicID  for the old style JIT intrinsics only: https://github.com/dotnet/coreclr/pull/14566/commits/f5a2b2d3edba440606375e766adcfaa3544cd94b